src/expandCollapseUtilities.js

```diff
@@ -783,7 +801,33 @@ return {
       edge.trigger('expandcollapse.beforeexpandedge');
       result.oldEdges = result.oldEdges.add(edge);
       cy.remove(edge);
-      result.edges = cy.add(edges);
```

Comment:
This is the main fix:
1) added try / catch to avoid breaking everything in case of an exception
2) before adding the new edges (after expanding a collapsed edge) verify that their source/target are visible
3) If source or target not visible treat the edge as a meta edge
  - save original ends
  - add meta edge class
  - instead of "move" just update source and data - move will fail since its source or target is missing
  
4) After everything call "repairEdge" to properly re-link all meta edges